satpy/readers/sar_c_safe.py

```diff
@@ -650,16 +627,19 @@ def _get_lonlatalts_uncached(self):
         Returns:
            coordinates (tuple): A tuple with longitude and latitude arrays
         """
-        band = self.filehandle
+        shape = self._data.shape
 
         (xpoints, ypoints), (gcp_lons, gcp_lats, gcp_alts), (gcps, crs) = self.get_gcps()
 
-        # FIXME: do interpolation on cartesian coordinates if the area is
-        # problematic.
+        fine_points = [np.arange(size) for size in shape]
+        x, y, z = lonlat2xyz(gcp_lons, gcp_lats)
+        interpolator = MultipleGridInterpolator((ypoints, xpoints), x, y, z, gcp_alts)
+        hx, hy, hz, altitudes = interpolator.interpolate(fine_points, method="cubic", chunks=self.chunks)
```

Comment:
With the newest scipy release, to get the same results as before you'll need to specify method="cubic_legacy" I think. Or update the tests to use the new "cubic" implementation which is supposed to be faster in some cases if I'm understanding the release notes correctly.

Reply:
I'm going for the new method I think